Sources/BSON/CodingStrategies.swift

```diff
@@ -0,0 +1,218 @@
+import Foundation
+
+/// Protocol indicating a set of options can be used to configure `BSONEncoder` and `BSONDecoder`.
+public protocol CodingStrategyProvider {
```

Comment:
I kind of wonder if this entire file still belongs in the driver itself, as this is all meant to support setting strategies for clients/dbs/collections. do you think it has any value for general BSON usage?

Reply:
I think it's still valuable in BSON-only contexts, mostly just to ensure that the strategies are consistent between the encoders/decoders. With e.g. `BSONCoderOptions`, a user could define a single options struct that they pass to both their `BSONEncoder()` and `BSONDecoder()` calls that ensures they both speak the same format.

Reply:
sgtm